src/cargo/ops/cargo_check.rs

```diff
@@ -0,0 +1,89 @@
+use core::Workspace;
+use ops::{self, CompileOptions, MessageFormat};
+use util::important_paths::{find_root_manifest_for_wd};
+use util::{CliResult, Config};
+
+#[derive(RustcDecodable)]
+pub struct Options {
```

Comment:
These structures typically are command-specific, so I'd prefer to leave this in `src/bin/check.rs`. The functionality though should all be configurable, I believe, through the options generated later. Was something missing though perhaps?